test/simplify_qdq_test.cpp

```diff
@@ -0,0 +1,484 @@
+#include <migraphx/simplify_qdq.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/ref/target.hpp>
+#include <migraphx/instruction.hpp>
+#include <test.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/matcher.hpp>
+
+bool is_convolution(const migraphx::instruction& ins) { return ins.name() == "convolution"; }
+bool is_dot(const migraphx::instruction& ins) { return ins.name() == "dot"; }
+
+static migraphx::shape s1{migraphx::shape::int8_type, {1}};
+static migraphx::shape s2{migraphx::shape::int8_type, {1280, 1000}};
+static migraphx::shape s3{migraphx::shape::int8_type, {1000}};
+static migraphx::shape s4{migraphx::shape::int8_type, {1280, 320, 1, 1}};
+static migraphx::shape s5{migraphx::shape::int32_type, {1}};
+static migraphx::shape s6{migraphx::shape::int32_type, {1280}};
+static migraphx::shape s7{migraphx::shape::float_type, {1, 320, 7, 7}};
+static migraphx::shape s8{migraphx::shape::float_type, {1}};
+
+static std::vector<std::vector<float>> floats{{0.00122128},
+                                              {0.168607},
+                                              {0.00260497},
+                                              {0.0235534},
+                                              {0.0235534},
+                                              {0.023622},
+                                              {0.0162265},
+                                              {0.0138147}};
+static std::vector<std::vector<int>> ints{{-35}, {-127}, {-18}, {0}};
+
+TEST_CASE(conv)
+{
+    migraphx::module m1;
+    {
+        auto pa  = m1.add_parameter("pa", s7);
+        auto p3  = m1.add_parameter("p3", s4);
+        auto l4  = m1.add_literal(s1, ints[1]);
+        auto l5  = m1.add_literal(s1, ints[2]);
+        auto l7  = m1.add_literal(s1, ints[3]);
+        auto l14 = m1.add_literal(s8, floats[5]);
+        auto l15 = m1.add_literal(s8, floats[6]);
+        auto l16 = m1.add_literal(s8, floats[7]);
+
+        auto mb1 = m1.add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1280, 320, 1, 1}}}), l16);
+        auto mb2 = m1.add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1280, 320, 1, 1}}}), l7);
+        auto d1  = m1.add_instruction(migraphx::make_op("dequantizelinear"), p3, mb1, mb2);
```

Comment:
I think it would be good to have a function to add these operators to make the test a little more concise:

```cpp
instruction_ref add_quantize_op(module& m, const std::string& name, instruction_ref x, instruction_ref scale, instruction_ref shift)
{
    auto lens = x->get_shape().lens();
    auto scale_mb = m.add_instruction(make_op("multibroadcast", {{"output_lens", lens}}), scale);
    auto shift_mb = m.add_instruction(make_op("multibroadcast", {{"output_lens", lens}}), shift);
    return m.add_instruction(make_op(name), x, scale_mb, shift_mb);
}
```

Then you can add the op with `auto d1 = add_quantize_op("dequantizelinear"), p3, scale1, shift1);` and dont have to worry about inserting the multibroadcasts.

Reply:
Also the variable should be named something like `dq_p3`(or `dq_input`), so it easier to see how it was transformed.